### PR TITLE
nautilus: msgr: backport the EventCenter-related fixes

### PR DESCRIPTION
**Pull request by rzarzynski**
backport trackers:

* https://tracker.ceph.com/issues/42856
* https://tracker.ceph.com/issues/43775
* https://tracker.ceph.com/issues/43832
* https://tracker.ceph.com/issues/44516

---

backport of 

* https://github.com/ceph/ceph/pull/27094
* https://github.com/ceph/ceph/pull/30717
* https://github.com/ceph/ceph/pull/32352
* https://github.com/ceph/ceph/pull/32732
* https://github.com/ceph/ceph/pull/33742

parent trackers:

* https://tracker.ceph.com/issues/42026
* https://tracker.ceph.com/issues/43070
* https://tracker.ceph.com/issues/43667
* https://tracker.ceph.com/issues/43808

---

Derogates: https://github.com/ceph/ceph/pull/32922.
Backport ticket: https://tracker.ceph.com/issues/44516.

### CODE REVIEW

**Comment by smithfarm** [comment]
@rzarzynski By "derogates" did you mean it replaces https://github.com/ceph/ceph/pull/32922 ? I.e. can we close https://github.com/ceph/ceph/pull/32922 and update the relevant backport tracker tickets?

**Comment by rzarzynski** [comment]
@smithfarm: yup, I think we can close #32922.

**Comment by smithfarm** [comment]
```
167/167 Test  #51: unittest_lockdep ........................***Timeout 3600.01 sec
NDEBUG is NOT defined
did not load config file, using default settings.
[==========] Running 2 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 2 tests from lockdep
[ RUN      ] lockdep.abba

[WARNING] /home/jenkins-build/build/workspace/ceph-pull-requests/src/googletest/googletest/src/gtest-death-test.cc:836:: Death tests use fork(), which is unsafe particularly in a threaded context. For this test, Google Test detected 3 threads.


99% tests passed, 1 tests failed out of 167

Total Test time (real) = 3630.53 sec

The following tests FAILED:
	 51 - unittest_lockdep (Timeout)
```

**Comment by rzarzynski** [comment]
@smithfarm: I was able to observe this on `master`. See: https://tracker.ceph.com/issues/44447.

**Comment by smithfarm** [comment]
jenkins test make check

**Comment by yuriw** [comment]
[wip-yuri-testing-2020-03-09-1935-nautilus](https://trello.com/c/eClkQRUQ/840-wip-yuri-testing-2020-03-09-1935-nautilus)

**Comment by yuriw** [comment]
@liewegas @neha-ojha this passed tests and ready for merge so pls approve

**Comment by jc-lab** [comment]
https://tracker.ceph.com/issues/45523

I am the author of the above issue.

The PR seems to be merged with nautilus. But if you look at the history of v14.2.8 or v14.2.9 tag, there is no corresponding commit. Has the PR been released yet?

I am confused because the history of 14.2.9 tag is different from the history of nautilus. (I'm git newbie)

**Comment by smithfarm** [comment]
> I am confused because the history of 14.2.9 tag is different from the history of nautilus. (I'm git newbie)

Don't worry - this is confusing even for seasoned experts ;-)

The reason for this seeming contradiction is the fact that 14.2.9 was a so-called "hotfix" release. In a nutshell, here's what happens with a hotfix release (and what happened in this case):

1. 14.2.8 got released
2. a bunch of PRs got merged into nautilus on top of the v14.2.8 tag
3. 14.2.9 hotfix release was published - this was just one fix on top of `v14.2.8` tag - i.e. without the "bunch of PRs" that were waiting for release at the time
4. 14.2.9 hotfix release (including `v14.2.9` tag) merged into `nautilus`

As a result, if you run `git describe` on the merge commit of this PR, it says `v14.2.8-97-gd18a0ab1ecd`, which would seem to indicate that the PR was released in 14.2.9, but that's not the case. In fact, this PR has not yet been released, but it will be a part of the next regular (non-"hotfix") nautilus release.

TL;DR: when in doubt, refer to the release notes.

**Comment by jc-lab** [comment]
@smithfarm Really thank you very much! :>

**Comment by isi-lincoln** [comment]
We have also experience this issue a few times (manager crashing) on 14.2.9 and were wondering if there is a timeline on for when the next nautilus debian release will be?
